app/controllers/oaisys/pmh_controller.rb

```diff
@@ -17,59 +17,75 @@ class Oaisys::PMHController < Oaisys::ApplicationController
+    expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
   end
 
   # get_record is referring to the verb, not a getter.
   # rubocop:disable Naming/AccessorMethodName
   def get_record
-    expect_args for_verb: :GetRecord, required: [:identifier, :metadataPrefix]
+    expect_args required: [:identifier, :metadataPrefix]
   end
   # rubocop:enable Naming/AccessorMethodName
 
+  # TODO: Handle from, until, and resumptionToken arguments.
   def list_identifiers
-    expect_args for_verb: :ListIdentifiers, required: [:metadataPrefix], optional: [:from, :until, :set],
-                exclusive: [:resumptionToken]
+    parameters = expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
+    identifiers = if params[:set].present?
+                    public_items_for_metadata_format(verb: parameters[:verb], format: params[:metadataPrefix],
+                                                     restricted_to_set: params[:set])
+                      .pluck(:id, :record_created_at, :member_of_paths)
+                  else
+                    public_items_for_metadata_format(verb: parameters[:verb], format: params[:metadataPrefix])
+                      .pluck(:id, :record_created_at, :member_of_paths)
+                  end
+    raise Oaisys::NoRecordsMatchError.new(parameters: parameters.slice(:verb, :metadataPrefix)) if identifiers.empty?
+
+    respond_to do |format|
+      format.xml do
+        render :list_identifiers, locals: { identifiers: identifiers, parameters: parameters }
+      end
+    end
   end
 
   private
 
-  def expect_args(for_verb:, required: [], optional: [], exclusive: [])
-    arguments = params.except('verb', 'controller', 'action').keys.map(&:to_sym)
-    expected_verb_arguments = required + optional + exclusive
-    unexpected_arguments = (arguments - expected_verb_arguments).present?
-    missing_required_arguments = (required - arguments).present?
-
-    raise Oaisys::BadArgumentError.new(for_verb: for_verb) if unexpected_arguments || missing_required_arguments
+  def expect_args(required: [], optional: [], exclusive: [])
+    params.require([:verb] + required)
+    params.permit([:verb] + required + optional + exclusive).to_h
   end
 
-  # Conventional way of calling expect_args for a verb with no arguments.
-  def expect_no_args(for_verb:)
-    expect_args(for_verb: for_verb)
+  def public_items_for_metadata_format(verb:, format:, restricted_to_set: nil)
+    model = case format
+            when 'oai_dc'
+              Oaisys::Engine.config.oai_dc_model
+            when 'oai_etdms'
+              Oaisys::Engine.config.oai_etdms_model
+            else
+              raise Oaisys::CannotDisseminateFormatError.new(parameters: { verb: verb, metadataPrefix: format })
+            end
+
+    !restricted_to_set.nil? ? model.public_items.belongs_to_path(restricted_to_set.tr(':', '/')) : model.public_items
```

Comment:
Instead of `!restricted_to_set.nil?`, `restricted_to_set.present?` is the same thing without the double-negative logic

Reply:
I've made a change in the other PR that now does `unless restricted_to_set.nil?`so I'll change it to `if restricted_to_set.present?`